jdbi/src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -108,22 +101,20 @@ public String locate(String name, StatementContext ctx) throws Exception
             }
 
             if (in_stream == null && ctx.getSqlObjectType() != null) {
-                String filename = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
+                final String filename = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
                 in_stream = loader.getResourceAsStream(filename);
                 if (in_stream == null) {
                     in_stream = ctx.getSqlObjectType().getResourceAsStream(filename);
                 }
             }
 
             if (in_stream == null) {
-                // Ensure we don't store an identity map entry which has a hard reference
-                // to the key (through the value) by copying the value, avoids potential memory leak.
-                found.put(cache_key, name == cache_key ? new String(name) : name);
+                found.put(cacheKey, name);
```

Comment:
Are you sure about this? See https://github.com/jdbi/jdbi/issues/93 and https://github.com/jdbi/jdbi/pull/125.